rio_cogeo/scripts/cli.py

```diff
@@ -279,6 +269,21 @@ def create(
     elif web_optimized:
         tilematrixset = morecantile.tms.get("WebMercatorQuad")
 
+    if tilematrixset:
+        blocksize = blocksize or min(
+            tilematrixset.matrix(tilematrixset.minzoom).tileHeight,
+            tilematrixset.matrix(tilematrixset.minzoom).tileWidth,
+        )
+        overview_blocksize = overview_blocksize or blocksize
```

Comment:
Personnally, I would echo a warning if conflicting parameters were received, and adapt blocksize/overview_blocksize to the tms max/min levels respectively:

```suggestion
    if tilematrixset:
        if blocksize:
            click.secho(
                f"`blocksize` argument will override provided `tms` (or web_optimized) setting",
                fg="yellow"
            )
        else:
            high_res_matrix = tilematrixset.matrix(tilematrixset.maxzoom)
            blocksize = min(high_res_matrix.tileHeight, high_res_matrix.tileWidth)

        if overview_blocksize:
            click.secho(
                f"`overview_blocksize` argument will override provided `tms` (or web_optimized) setting",
                fg="yellow"
            )
        else:
            low_res_matrix = tilematrixset.matrix(tilematrixset.minzoom)
            overview_blocksize = min(low_res_matrix.tileHeight, low_res_matrix.tileWidth)

```

Otherwise, it looks all good for me.